pipelines/batch/src/main/java/org/openmrs/analytics/ConvertResourceFn.java

```diff
@@ -0,0 +1,135 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.openmrs.analytics;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.parser.IParser;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.beam.sdk.metrics.Counter;
+import org.apache.beam.sdk.metrics.Metrics;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.hl7.fhir.r4.model.Meta;
+import org.hl7.fhir.r4.model.Resource;
+
+public class ConvertResourceFn extends DoFn<List<String>, Integer> {
+	
+	private final HashMap<String, Counter> numFetchedResources;
+	
+	private final HashMap<String, Counter> totalParseTimeMillis;
+	
+	private final HashMap<String, Counter> totalGenerateTimeMillis;
+	
+	private final HashMap<String, Counter> totalPushTimeMillis;
+	
+	private final String sinkPath;
+	
+	private final String sinkUsername;
+	
+	private final String sinkPassword;
+	
+	private final String parquetFile;
+	
+	private final int secondsToFlush;
+	
+	private final int rowGroupSize;
+	
+	@VisibleForTesting
+	protected ParquetUtil parquetUtil;
+	
+	protected FhirStoreUtil fhirStoreUtil;
+	
+	private SimpleDateFormat simpleDateFormat;
+	
+	private IParser parser;
+	
+	ConvertResourceFn(FhirEtlOptions options) {
+		this.sinkPath = options.getFhirSinkPath();
```

Comment:
A good amount of functionality/setup here is shared with `FetchSearchPageFn`. I think I understand why you have chosen not to use that as a base class but still there is room for refactoring. This does not need to happen in this PR; just add a TODO with reference to `FetchSearchPageFn` to do this in future.